File: ocaml/ex3_2_2.ml
type 'a chooser = Chooser of 'a * 'a

let choose (Chooser (v1, v2)) = if true then v1 else v2

let _ =
  let c: int chooser = Chooser (1, 2) in
  let n: int = choose c in
  ()

File: cpp/ex3_2_2.cpp
template <typename T> class Chooser {
public:
    T v1;
    T v2;
    T choose() { return true ? v1 : v2; }
};

int main() {
    Chooser<int> *c = new Chooser<int>();
    int n = c->choose();
}
